api/types/container/config.go

```diff
@@ -73,10 +73,11 @@ type Config struct {
 	// Mac Address of the container.
 	//
 	// Deprecated: this field is deprecated since API v1.44. Use EndpointSettings.MacAddress instead.
-	MacAddress  string            `json:",omitempty"`
-	OnBuild     []string          // ONBUILD metadata that were defined on the image Dockerfile
-	Labels      map[string]string // List of labels set to this container
-	StopSignal  string            `json:",omitempty"` // Signal to stop a container
-	StopTimeout *int              `json:",omitempty"` // Timeout (in seconds) to stop a container
-	Shell       strslice.StrSlice `json:",omitempty"` // Shell for shell-form of RUN, CMD, ENTRYPOINT
+	MacAddress    string            `json:",omitempty"`
+	OnBuild       []string          // ONBUILD metadata that were defined on the image Dockerfile
+	Labels        map[string]string // List of labels set to this container
+	StopSignal    string            `json:",omitempty"` // Signal to stop a container
+	StopTimeout   *int              `json:",omitempty"` // Timeout (in seconds) to stop a container
+	DeleteTimeout *int              `json:",omitempty"` // Timeout (in seconds) to wait for a container to be deleted
```

Comment:
Adding a new field here requires adjusting the image config comparison function used by the classic builder cache:
https://github.com/moby/moby/blob/b7c059886c0898436db90b4615a27cfb4d93ce34/image/cache/compare.go#L147-L151